after upgrading to angular 13 when run ng test --code-coverage  not working !!!

### Which @angular/* package(s) are the source of the bug?

compiler-cli, upgrade

### Is this a regression?

Yes

### Description

after upgrading to angular 13 when run ng test --code-coverage 
Error: Module not found: Error: Can't resolve 'core-js/proposals/reflect-metadata'

### Please provide a link to a minimal reproduction of the bug

_No response_

### Please provide the exception or error you saw

_No response_

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 13.1.2
Node: 14.15.1
Package Manager: npm 6.14.8
OS: win32 x64
```


### Anything else?

_No response_

I also got this error, though it wasn't after upgrading to Angular 13, it was after removing a package (devextreme) which depended on `core-js`. I also fixed it by adding `core-js` as a local dependency. In diagnosing this issue, I think the issue comes from [here](https://github.com/angular/angular-cli/blob/2c9a33dddb38694b6940ec6981c49904de1ab636/packages/angular_devkit/build_angular/src/webpack/configs/common.ts#L128). Which I think is not respecting the path of the transitive dependencies of the devkit package and is instead trying to find it in the context of the host project.

Some more detailed logs:
```
 ng test --watch=false
⠙ Generating browser application bundles (phase: setup)...07 01 2022 07:07:56.953:INFO [karma-server]: Karma v6.3.9 server started at http://localhost:9876/
07 01 2022 07:07:56.954:INFO [launcher]: Launching browsers FirefoxHeadless with concurrency unlimited
07 01 2022 07:07:56.957:INFO [launcher]: Starting browser FirefoxHeadless
07 01 2022 07:07:56.963:WARN [FirefoxHeadlessLauncher]: WSL environment detected: Please do not open Firefox while running tests as it will be killed after the test!
07 01 2022 07:07:56.963:WARN [FirefoxHeadlessLauncher]: WSL environment detected: See https://github.com/karma-runner/karma-firefox-launcher/issues/101#issuecomment-891850143
✔ Browser application bundle generation complete.

Error: Module not found: Error: Can't resolve 'core-js/proposals/reflect-metadata' in '<project_root>'
```

Before devextreme removal
```
npm ls core-js
├─┬ @angular-devkit/build-angular@13.1.2
│ └── core-js@3.19.3
└─┬ devextreme@20.2.10
  └─┬ devextreme-quill@1.1.5
    └── core-js@3.20.0
```

After devextreme removal
```
npm ls core-js
└─┬ @angular-devkit/build-angular@13.1.2
  └── core-js@3.19.3
```

And after the workaround:
```
orchestrate-web-client@2021.12 /home/julian/dev/orchestrate-platform/web-client
├─┬ @angular-devkit/build-angular@13.1.2
│ └── core-js@3.19.3
└── core-js@3.20.2
```

--- a/packages/angular_devkit/build_angular/src/webpack/configs/common.ts
+++ b/packages/angular_devkit/build_angular/src/webpack/configs/common.ts
@@ -125,7 +125,7 @@ export async function getCommonConfig(wco: WebpackConfigOptions): Promise<Config
     }
 
     if (!buildOptions.aot) {
-      const jitPolyfills = 'core-js/proposals/reflect-metadata';
+      const jitPolyfills = require.resolve('core-js/proposals/reflect-metadata');
       if (entryPoints['polyfills']) {
         entryPoints['polyfills'].push(jitPolyfills);
       } else {

